acceptance/chaos_test.go

```diff
@@ -113,27 +115,39 @@ func TestChaos(t *testing.T) {
 				return
 			default:
 			}
+			atomic.StoreInt64(&roundCopy, int64(round))
 			nodes := rnd.Perm(num)[:rnd.Intn(num)+1]
-
 			log.Infof("round %d: restarting nodes %v", round, nodes)
 			for i := 0; i < num; i++ {
 				clients[i].Lock()
 			}
-			for i := 0; i < num; i++ {
-				log.Infof("restarting %v", i)
+			for _, i := range nodes {
+				select {
+				case <-stopper:
+					break
+				default:
+				}
+				if time.Now().After(deadline) {
+					break
+				}
+				log.Infof("restarting %d", i)
 				c.Kill(i)
 				c.Restart(i)
 				initClient(i)
+			}
+			for i := 0; i < num; i++ {
```

Comment:
This was to ensure that if we exit the the previous loop, that we will still release all locks.